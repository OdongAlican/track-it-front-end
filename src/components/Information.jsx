import React from 'react'
import Footer from './Footer'

const Information = () => (
    <div >
    <div className="information-section">
        <div className="copyright p-2 mb-3 d-flex ">
        <h5 className="mx-auto"> Copyright © 2020 by Odong Alican </h5>
        </div>
       <div className="p-2 text-center text-secondary copyright">
            All rights reserved. No part of this publication may be reproduced, distributed,
            or transmitted in any form or by any means, including photocopying, recording,
            or other electronic or mechanical methods, without the prior written permission of the developer
            , except in the case of brief quotations embodied in critical reviews and
            certain other noncommercial uses permitted by copyright law. 
        </div>
        <div className="copyright text-center p-4 text-secondary">
            For permission requests,
            write to the Software developer, addressed “Attention: Permissions Coordinator,” at the address below.
            <p>sandieo.2020@gmail.com</p>
        </div>
    </div>
    <Footer/>
    </div>
)

export default Information