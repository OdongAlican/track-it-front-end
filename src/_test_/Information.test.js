import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import Information from '../components/Information';

afterEach(cleanup);
const initialState = {
  authReducer: {
    output: 10,
    error: 'test',
  },
};

const mockStore = configureStore();
let store;

it('renders Information correctly', () => {
  store = mockStore(initialState);
  const { getAllByTestId } = render(
    <Router>
      <Provider store={store}><Information /></Provider>
    </Router>,
  );
  const result = getAllByTestId('appInformation');
  expect(result[0])
    .toHaveTextContent('HomeCreate ActivityInformationMore');
});

it('renders Information correctly', () => {
  store = mockStore(initialState);
  const { getAllByTestId } = render(
    <Router>
      <Provider store={store}><Information /></Provider>
    </Router>,
  );
  const result = getAllByTestId('appInformation');
  expect(result[0]).not
    .toHaveTextContent('Some Random Text');
});

it('matches Information snapshot', () => {
  const appointment = renderer.create(
    <Router>
      <Provider store={store}><Information /></Provider>
    </Router>,
  );
  const tree = appointment.toJSON();
  expect(tree).toMatchSnapshot();
});
