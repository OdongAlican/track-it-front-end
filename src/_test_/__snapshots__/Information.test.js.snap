// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches Information snapshot 1`] = `
<div
  data-testid="appInformation"
>
  <div
    className="information-section"
  >
    <div
      className="copyright p-2 mb-3 d-flex "
    >
      <h5
        className="mx-auto"
      >
         Copyright © 2020 by Odong Alican 
      </h5>
    </div>
    <div
      className="p-2 text-center text-secondary copyright"
    >
      All rights reserved. No part of this publication may be reproduced, distributed, or transmitted in any form or by any means, including photocopying, recording, or other electronic or mechanical methods, without the prior written permission of the developer , except in the case of brief quotations embodied in critical reviews and certain other noncommercial uses permitted by copyright law.
    </div>
    <div
      className="copyright text-center p-4 text-secondary"
    >
      For permission requests, write to the Software developer, addressed “Attention: Permissions Coordinator,” at the address below.
      <p>
        sandieo.2020@gmail.com
      </p>
    </div>
  </div>
  <div
    className="footer"
    data-testid="appFooter"
  >
    <a
      aria-current={null}
      className="nav-link"
      href="/activities"
      onClick={[Function]}
      style={
        Object {
          "color": "#ABAEB3",
          "padding": 0,
          "textDecoration": "none",
        }
      }
    >
      <div
        className="link-div"
        data-testid="appButton"
      >
        <i
          className="far fa-chart-bar footer-btn"
        />
        <p>
          Home
        </p>
      </div>
    </a>
    <a
      aria-current={null}
      className="nav-link"
      href="/create-activity"
      onClick={[Function]}
      style={
        Object {
          "color": "#ABAEB3",
          "padding": 0,
          "textDecoration": "none",
        }
      }
    >
      <div
        className="link-div"
        data-testid="appButton"
      >
        <i
          className="fas fa-chart-line footer-btn"
        />
        <p>
          Create Activity
        </p>
      </div>
    </a>
    <a
      aria-current={null}
      className="nav-link"
      href="/information"
      onClick={[Function]}
      style={
        Object {
          "color": "#ABAEB3",
          "padding": 0,
          "textDecoration": "none",
        }
      }
    >
      <div
        className="link-div"
        data-testid="appButton"
      >
        <i
          className="fas fa-chart-pie footer-btn"
        />
        <p>
          Information
        </p>
      </div>
    </a>
    <a
      aria-current={null}
      className="nav-link"
      href="/activities"
      onClick={[Function]}
      style={
        Object {
          "color": "#ABAEB3",
          "padding": 0,
          "textDecoration": "none",
        }
      }
    >
      <div
        className="link-div"
        data-testid="appButton"
      >
        <i
          className="fas fa-ellipsis-h footer-btn"
        />
        <p>
          More
        </p>
      </div>
    </a>
  </div>
</div>
`;
